File: src/modelos/Jogador.ts
import { ListaEncadeadaSimples } from "../estruturas/ListaEncadeadaSimple";
import { Carta } from "./Carta"

export class Jogador {
    id: number;
    nome: string;
    isBot: boolean;
    mao: ListaEncadeadaSimples<Carta> = new ListaEncadeadaSimples();

    constructor(id: number, nome: string, isBot: boolean) {
        this.id = id;
        this.nome = nome;
        this.isBot = isBot;
        this.mao = new ListaEncadeadaSimples<Carta>();
    }
}

File: uno-server/src/main/java/com/usjt/a3/unogame/modelo/Jogador.java
package com.usjt.a3.unogame.modelo;

import com.usjt.a3.unogame.estrutura.ListaEncadeada;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Jogador {
    private Long id;
    private String nome;
    private boolean isBot = false;
    private ListaEncadeada<Carta> mao = new ListaEncadeada<>();

    public Jogador(Long id, String nome, Boolean isBot) {
        this.id = id;
        this.nome = nome;
        this.isBot = isBot;
    }
}
